grpc/src/jmh/java/com/linecorp/armeria/grpc/downstream/DownstreamSimpleBenchmark.java

```diff
@@ -17,58 +17,67 @@
 import com.linecorp.armeria.grpc.shared.GithubApiService;
+import com.linecorp.armeria.grpc.shared.SimpleBenchmarkBase;
 import com.linecorp.armeria.server.Server;
 import com.linecorp.armeria.server.ServerBuilder;
 import com.linecorp.armeria.server.ServerPort;
 import com.linecorp.armeria.server.grpc.GrpcServiceBuilder;
 
 @State(Scope.Benchmark)
-public class DownstreamSimpleBenchmark {
+public class DownstreamSimpleBenchmark extends SimpleBenchmarkBase {
 
     private Server server;
     private GithubServiceBlockingStub githubApiClient;
+    private GithubServiceFutureStub githubApiFutureClient;
 
-    @Setup
-    public void startServer() throws Exception {
-        server = new ServerBuilder()
-                .port(0, HTTP)
-                .serviceUnder("/", new GrpcServiceBuilder().addService(new GithubApiService()).build())
-                .build();
-        server.start().join();
+    @Override
+    protected int port() {
         ServerPort httpPort = server.activePorts().values().stream()
                                     .filter(p1 -> p1.protocol() == HTTP).findAny()
                                     .get();
-        githubApiClient = Clients.newClient(
-                "gproto+http://127.0.0.1:" + httpPort.localAddress().getPort() + "/",
-                GithubServiceBlockingStub.class);
+        return httpPort.localAddress().getPort();
     }
 
-    @TearDown
-    public void stopServer() throws Exception {
-        server.stop().join();
+    @Override
+    protected GithubServiceBlockingStub normalClient() {
+        return githubApiClient;
+    }
+
+    @Override
+    protected GithubServiceFutureStub normalFutureClient() {
+        return githubApiFutureClient;
     }
 
-    @Benchmark
-    public void simple(Blackhole bh) throws Exception {
-        bh.consume(githubApiClient.simple(SEARCH_RESPONSE));
+    @Override
+    protected void setUp() throws Exception {
+        server = new ServerBuilder()
+                .port(0, HTTP)
+                .serviceUnder("/", new GrpcServiceBuilder().addService(new GithubApiService()).build())
+                .build();
+        server.start().join();
+        final String url = "gproto+http://127.0.0.1:" + port() + "/";
+        githubApiClient = Clients.newClient(url, GithubServiceBlockingStub.class);
+        githubApiFutureClient = Clients.newClient(url, GithubServiceFutureStub.class);
     }
 
-    @Benchmark
-    public void empty(Blackhole bh) throws Exception {
-        bh.consume(githubApiClient.empty(Empty.getDefaultInstance()));
+    @Override
+    protected void tearDown() throws Exception {
+        server.stop().join();
     }
+
+    /*public static void main(String[] args) throws Exception {
+        DownstreamSimpleBenchmark benchmark = new DownstreamSimpleBenchmark();
+        benchmark.start();
+        System.out.println(benchmark.simple());
+        System.out.println(benchmark.empty());
+        benchmark.stop();
+    }*/
```

Comment:
Which would be better? `/* .. */` or `//`?

Reply:
Switched to line comment